lib/tasks/build_dependencies.rake

```diff
@@ -51,14 +52,23 @@ namespace :build_dependencies do
     Dir.glob("#{Packaging::Configuration.instance.package_dir}/*.spec").each do |spec_file|
       # replace the "@BUILD_FLAVOR@" placeholder by each flavor defined
       flavors.each do |flavor|
-        spec_content = File.read(spec_file).gsub("@BUILD_FLAVOR@", flavor)
+        spec_content = File.read(spec_file).gsub!("@BUILD_FLAVOR@", flavor)
 
-        # get the BuildRequires from the spec files, this also expands the RPM macros like %{rubygem}
-        # use Open3 as the command produces some bogus error messages on stderr even on success,
-        # but in case of error it provides a hint what failed
-        stdout, stderr, status = Open3.capture3("rpmspec", "-q", "--buildrequires", "/dev/stdin", stdin_data: spec_content)
-        raise "Parsing #{spec_file} (flavor #{flavor.inspect})failed:\n#{stderr}" unless status.success?
-        buildrequires.concat(stdout.split("\n"))
+        if spec_content.nil?
+          # no replacement, use the file directly
+          stdout = `rpmspec -q --buildrequires #{spec_file.shellescape}`
+          raise "Parsing #{spec_file} failed" unless $?.success?
+          buildrequires.concat(stdout.split("\n"))
+        else
+          # rpmspec can only read a file, write the processed data to a temporary file
+          Tempfile.create(["rake_build_deps-", ".spec"]) do |tmp|
+            tmp.write(spec_content)
+            tmp.flush
+            stdout = `rpmspec -q --buildrequires #{tmp.path.shellescape}`
```

Comment:
NP: I would like to factor out the duplication to `buildrequires_from_filename(filename)`. Or even always create the Tempfile for simplicity, the preformance hit does not matter in this usecase.